### PR TITLE
Use OTString to instantiate String objects: Encode.hpp

### PR DESCRIPTION
**Pull request by Wallydog66**
Passes all tests.

Signed-off-by: Wallydog66 <wallydog.66@gmail.com>

### CODE REVIEW

**Comment by Wallydog66** [comment]
*** DOES NOT PASS TESTS YET ***

File: include/opentxs/Proto.hpp [hunk 1]
```diff
@@ -91,7 +91,7 @@ template <class T>
 T StringToProto(const String& input)
 {
     Armored armored;
-    String unconstInput(input);
+    OTString unconstInput(input);
```
Comment (FellowTraveler) [inline include/opentxs/Proto.hpp]: Shouldn't this be: 
   OTString unconstInput = String::Factory(input);

File: src/client/OTAPI_Exec.cpp [hunk 2]
```diff
@@ -9893,7 +9936,7 @@ std::string OTAPI_Exec::Purse_Push(
                                                                      // purse is
     // password-protected, then this
     // parameter should be "".
-    strReason.empty();
+    strReason->empty();
```
Comment (FellowTraveler) [inline src/client/OTAPI_Exec.cpp]: This makes no sense.

strReason->empty(); <=== This just returns true/false based on whether or not the string is empty. Then we're at the end of the statement and that boolean value is simply discarded unused.

Probably meant to do strReason->clear();   ????  That will clear the string and MAKE it empty.

I know the bug was introduced in some previous PR (not this one) but it's definitely a bug.

File: src/client/OT_API.cpp [hunk 3]
```diff
@@ -9663,8 +9697,8 @@ CommandResult OT_API::triggerClause(
 
     // Optional string parameter. Available as "param_string" inside the
     // script.
-    if ((nullptr != pStrParam) && (pStrParam->Exists())) {
-        payload.SetString(*pStrParam);
+    if ((pStrParam.Exists()) && (pStrParam.Exists())) {
```
Comment (FellowTraveler) [inline src/client/OT_API.cpp]: Change this line:
    if ((pStrParam.Exists()) && (pStrParam.Exists())) {

To this:
   if (pStrParam.Exists()) {

(No need to repeat the same expression twice. The old code was checking the pointer for null, and then checking the string for Exists(). The new code is no longer using a raw pointer).

File: src/core/contract/ServerContract.cpp [hunk 4]
```diff
@@ -201,9 +201,9 @@ proto::ServerContract ServerContract::IDVersion(const Lock& lock) const
     contract.clear_publicnym();  // reinforcing that this field must be blank.
 
     if (nullptr != nym_) {
```
Comment (FellowTraveler) [inline src/core/contract/ServerContract.cpp]: Change:
   if (nullptr != nym_) {

To:
   if (nym_) {

Looks like this line never got fixed once the nym_ member variable was changed from a raw pointer to a smart pointer.

**Pull request merged**